Photo Collection/Photo Collection/Views/PhotosCollectionViewController.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  PhotosCollectionViewController.swift
+//  Photo Collection
+//
+//  Created by Angel Buenrostro on 1/16/19.
+//  Copyright © 2019 Angel Buenrostro. All rights reserved.
+//
+
+import UIKit
+
+
+class PhotosCollectionViewController: UICollectionViewController {
+    
+    let photoController = PhotoController()
+    let themeHelper = ThemeHelper()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setTheme()
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+
+        // Register cell classes
+        
+
+        // Do any additional setup after loading the view.
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        setTheme()
+        self.collectionView.reloadData()
+    }
+
+    
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using [segue destinationViewController].
+        switch segue.identifier {
+        case "themeSegue":
+             let themeVC = segue.destination as! ThemeSelectionViewController
+             themeVC.themeHelper = themeHelper
+        case "createSegue":
+            let detailVC = segue.destination as! PhotoDetailViewController
+            detailVC.photoController = photoController
+            detailVC.themeHelper     = themeHelper
+           
+        case "cellSegue":
+            let cellVC = segue.destination as! PhotoDetailViewController
+            let cell = sender as! UICollectionViewCell
+            let indexPath = collectionView.indexPath(for: cell)
+            cellVC.photoController  = photoController
+            cellVC.themeHelper      = themeHelper
+            cellVC.photo            = photoController.photos[indexPath!.item]
+        default:
+             fatalError("segue could not be prepared for")
+        }
+        // Pass the selected object to the new view controller.
+    }
+    
+
+    // MARK: UICollectionViewDataSource
+
+
+
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of items
+        return photoController.photos.count
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "PhotoCell", for: indexPath)
+        guard let unwrappedCell = cell as? PhotosCollectionViewCell else { fatalError("No photo cell")}
+        
+        let photo = photoController.photos[indexPath.item]
+        unwrappedCell.imageView.image = UIImage(data: photo.imageData)
+        unwrappedCell.photoLabel.text = photo.title
```

Comment:
It's not a good practice to set custom cell's properties inside of `cellForItemAt IndexPath`, performance wise. And it violates MVC pattern. We pass the object and gives that responsibility to the cell. So remove line 76-77 and just pass `photo` object
```suggestion
        unwrappedCell.photo = photo
```